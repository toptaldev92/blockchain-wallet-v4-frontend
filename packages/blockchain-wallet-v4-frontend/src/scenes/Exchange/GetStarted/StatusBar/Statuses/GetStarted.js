import { Button, Text } from 'blockchain-info-components'
import { FormattedMessage } from 'react-intl'
import { LinkContainer } from 'react-router-bootstrap'
import React from 'react'
import styled from 'styled-components'

const ActionButton = styled(Button).attrs({ nature: 'primary' })`
  font-weight: 600;
  width: 264px;
  height: 48px;
`

const GetStarted = () => (
  <LinkContainer to='/swap/profile'>
    <ActionButton>
      <Text color='white' size='16px' weight={600} data-e2e='swapGetStarted'>
        <FormattedMessage
          id='scenes.exchange.getstarted.status.getstarted.button'
          defaultMessage='Get Started'
        />
      </Text>
    </ActionButton>
  </LinkContainer>
)

export default GetStarted
