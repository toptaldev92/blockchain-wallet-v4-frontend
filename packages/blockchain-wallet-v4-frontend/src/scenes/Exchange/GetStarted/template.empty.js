import { Box } from 'components/Box'
import { FormattedMessage } from 'react-intl'
import { Image, Text } from 'blockchain-info-components'
import EmptySceneCarousel from 'components/EmptySceneCarousel'
import React from 'react'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
`

const Slide = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  & > :first-child {
    margin-bottom: 10px;
  }
  & > :last-child {
    align-self: flex-end;
    justify-self: flex-end;
  }
`
const EmptySwapCarousel = () => {
  return (
    <Wrapper>
      <Box>
        <EmptySceneCarousel height={230}>
          <Slide>
            <Text size='20px' weight={600} color='grey800'>
              <FormattedMessage
                id='swap.carousel.title.page2'
                defaultMessage='Real-time Exchange Rates'
              />
            </Text>
            <Text size='14px' weight={500} color='grey600' lineHeight='20px'>
              <FormattedMessage
                id='swap.carousel.desc.page2'
                defaultMessage='Access competitive crypto prices right at your fingertips.'
              />
            </Text>
            <Image name='swap-carousel2' width='170px' />
          </Slide>
          <Slide>
            <Text size='20px' weight={600} color='grey800'>
              <FormattedMessage
                id='swap.carousel.title.page3'
                defaultMessage='100% On-Chain'
              />
            </Text>
            <Text size='14px' weight={500} color='grey600' lineHeight='20px'>
              <FormattedMessage
                id='swap.carousel.desc.page3'
                defaultMessage='All Swap trades are confirmed and settled directly on-chain.'
              />
            </Text>
            <Image name='swap-carousel3' width='170px' />
          </Slide>
          <Slide>
            <Text size='20px' weight={600} color='grey800'>
              <FormattedMessage
                id='swap.carousel.title.page4'
                defaultMessage='You Control Your Key'
              />
            </Text>
            <Text size='14px' weight={500} color='grey600' lineHeight='20px'>
              <FormattedMessage
                id='swap.carousel.desc.page4'
                defaultMessage='With Swap your crypto is safe, secure, and your keys are always intact.'
              />
            </Text>
            <Image name='swap-carousel4' width='170px' />
          </Slide>
          <Slide>
            <Text size='20px' weight={600} color='grey800'>
              <FormattedMessage
                id='swap.carousel.title.page5'
                defaultMessage='Manage Risk Better'
              />
            </Text>
            <Text size='14px' weight={500} color='grey600' lineHeight='20px'>
              <FormattedMessage
                id='swap.carousel.desc.page5-1'
                defaultMessage='Introducing Digital Dollars (USD Digital) to de-risk your crypto investment or lock-in gains.'
              />
            </Text>
            <Image name='swap-carousel5' width='170px' />
          </Slide>
        </EmptySceneCarousel>
      </Box>
      <Box />
    </Wrapper>
  )
}

export default EmptySwapCarousel
