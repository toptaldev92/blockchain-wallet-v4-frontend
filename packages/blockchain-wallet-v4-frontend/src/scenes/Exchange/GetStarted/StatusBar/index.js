import React from 'react'

import GetStarted from './Statuses/GetStarted'
import InProgress from './Statuses/InProgress'
import Pending from './Statuses/Pending'
import Rejected from './Statuses/Rejected'
import UnderReview from './Statuses/UnderReview'

const Status = ({ step }) => (
  <>
    {step === 'getstarted' && <GetStarted />}
    {step === 'inprogress' && <InProgress />}
    {step === 'pending' && <Pending />}
    {step === 'underreview' && <UnderReview />}
    {step === 'rejected' && <Rejected />}
  </>
)

export default Status
