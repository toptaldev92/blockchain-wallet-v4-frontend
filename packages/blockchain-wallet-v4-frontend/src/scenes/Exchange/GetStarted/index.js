import 'slick-carousel/slick/slick-theme.css'
import 'slick-carousel/slick/slick.css'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import { getData } from './selectors'
import { Image, Text } from 'blockchain-info-components'
import media from 'services/ResponsiveService'
import React, { useState } from 'react'
import Slider from 'react-slick'
import StatusBar from './StatusBar'
import styled, { css } from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  max-width: 1000px;
  height: auto;
  padding: 25px 50px 50px;
  box-sizing: border-box;
  margin: 0 auto;

  ${media.mobile`
    padding: 0;
  `};
`

const Row = styled.div`
  display: flex;
`

const Column = styled.div`
  display: flex;
  flex-direction: column;
`

const CarouselImgCss = css`
  position: absolute;
  right: 0;
  top: 80px;
  width: 304px;
  transition: opacity 0.4s ease-in-out;
  background: ${({ theme }) => theme.white};
  z-index: 2;

  ${media.laptop`
    top: 0;
    right: 0;
    left: 0;
    margin: 0 auto;
  `}

  ${media.mobile`
    width: 272px;
  `}
`

const NextButton = styled.div`
  &.slick-next {
    height: 48px;
    width: 48px;
    border-bottom: 2px solid ${props => props.theme.blue600};
    border-right: 2px solid ${props => props.theme.blue600};
    transform: rotate(-45deg);
    right: -80px;
    border-bottom-right-radius: 8px;
    display: ${({ lastPage }) => (lastPage ? 'none' : 'block')};

    ${media.mobile`
      display: none;
    `};
  }
`

const PrevButton = styled.div`
  &.slick-prev {
    height: 48px;
    width: 48px;
    border-bottom: 2px solid ${props => props.theme.blue600};
    border-left: 2px solid ${props => props.theme.blue600};
    transform: rotate(45deg);
    left: -80px;
    border-bottom-left-radius: 8px;
    display: ${({ firstPage }) => (firstPage ? 'none' : 'block')};

    ${media.mobile`
      display: none;
    `};
  }
`

const CarouselWrapper = styled(Column)`
  width: 80%;
  position: relative;
  align-items: flex-start;

  .slick-slider {
    width: 100%;
  }

  .slick-slider > ul {
    bottom: 125px;
    width: fit-content;

    ${media.laptop`
      width: 100%;
    `}
  }

  .slick-slider > ul > li:first-child {
    margin-left: 0;
  }

  .slick-slider > ul > li > a > div {
    background: ${({ theme }) => theme['gray-2']};
    ${media.laptop`
      background: ${({ theme }) => theme['l']};
    `}
  }

  .slick-slider > ul > li.slick-active > a > div {
    background: ${({ theme }) => theme.blue600};
  }

  .slick-dots li button:before {
    width: 24px;
    height: 4px;
  }
`

const CarouselPageWrapper = styled(Row)`
  ${media.laptop`
    justify-content: center;
  `}
`

const Container = styled(Column)`
  width: 50%;
  height: 468px;
  justify-content: center;

  ${media.laptop`
    width: 100%;
    margin-top: 96px;
    text-align: center;
  `};
`

const Title = styled(Text)`
  margin-bottom: 10px;
  color: ${({ theme }) => theme.blue900};
`

const BodyText = styled(Text)`
  padding-right: 15px;

  ${media.laptop`
    padding-right: 0;
  `}
`

const CarouselImage1 = styled(Image)`
  ${CarouselImgCss}
  opacity: ${({ currentSlide }) => (currentSlide === 0 ? 1 : 0)};
`

const CarouselImage2 = styled(Image)`
  ${CarouselImgCss}
  opacity: ${({ currentSlide }) => (currentSlide === 1 ? 1 : 0)};
`

const CarouselImage3 = styled(Image)`
  ${CarouselImgCss}
  opacity: ${({ currentSlide }) => (currentSlide === 2 ? 1 : 0)};
`

const CarouselImage4 = styled(Image)`
  ${CarouselImgCss}
  opacity: ${({ currentSlide }) => (currentSlide === 3 ? 1 : 0)};
`

const CarouselImage5 = styled(Image)`
  ${CarouselImgCss}
  opacity: ${({ currentSlide }) => (currentSlide === 4 ? 1 : 0)};
`

const Dots = styled.div`
  height: 4px;
  width: 24px;
  border-radius: 14px;

  ${media.laptop`
    width: 8px;
    height: 8px;
    border-radius: 50%;
  `}
`

const PlacementDiv = styled.div`
  position: absolute;
  right: 0;
  top: 0;
  width: 50%;
  height: 100%;
  background: ${({ theme }) => theme.white};
  z-index: 1;

  ${media.laptop`
    display: none;
 `}
`

const ButtonContainer = styled(Row)`
  width: 100%;
  margin-top: -72px;
  z-index: 1;
  margin-bottom: 64px;
  ${media.laptop`
    justify-content: center;
  `}
`

function NextArrow (props) {
  const { className, onClick, lastPage } = props
  return (
    <NextButton className={className} onClick={onClick} lastPage={lastPage} />
  )
}

function PrevArrow (props) {
  const { className, onClick, firstPage } = props
  return (
    <PrevButton className={className} onClick={onClick} firstPage={firstPage} />
  )
}

const GetStarted = ({ step }) => {
  const [currentSlide, setCurrentSlide] = useState(0)
  const [firstPage, setFirstPage] = useState(true)
  const [lastPage, setLastPage] = useState(false)

  const settings = {
    customPaging: function (i) {
      return (
        <a>
          <Dots />
        </a>
      )
    },
    focusOnSelect: false,
    dots: true,
    dotsClass: 'slick-dots slick-thumb',
    infinite: false,
    speed: 400,
    slidesToShow: 1,
    slidesToScroll: 1,
    beforeChange: (current, next) => {
      setCurrentSlide(next)
      if (current === 0 && firstPage) {
        setFirstPage(false)
      } else if (current === 4 && lastPage) {
        setLastPage(false)
      }
    },
    afterChange: current => {
      if (current === 0 && !firstPage) {
        setFirstPage(true)
      } else if (current === 4 && !lastPage) {
        setLastPage(true)
      }
    },
    nextArrow: <NextArrow lastPage={lastPage} />,
    prevArrow: <PrevArrow firstPage={firstPage} />
  }

  return (
    <Wrapper>
      <CarouselWrapper>
        <Slider {...settings}>
          <div>
            <CarouselPageWrapper>
              <Container>
                <Title size='24px' weight={600}>
                  <FormattedMessage
                    defaultMessage='Welcome to Swap!'
                    id='swap.carousel.title.page1'
                  />
                </Title>
                <BodyText weight={500}>
                  <FormattedMessage
                    defaultMessage='The easiest way to exchange one crypto for another without leaving your wallet.'
                    id='swap.carousel.desc.page1'
                  />
                </BodyText>
              </Container>
            </CarouselPageWrapper>
          </div>
          <div>
            <CarouselPageWrapper>
              <Container>
                <Title size='24px' weight={600}>
                  <FormattedMessage
                    defaultMessage='Real-time Exchange Rates'
                    id='swap.carousel.title.page2'
                  />
                </Title>
                <BodyText weight={500}>
                  <FormattedMessage
                    defaultMessage='Access competitive crypto prices right at your fingertips.'
                    id='swap.carousel.desc.page2'
                  />
                </BodyText>
              </Container>
            </CarouselPageWrapper>
          </div>
          <div>
            <CarouselPageWrapper>
              <Container>
                <Title size='24px' weight={600}>
                  <FormattedMessage
                    defaultMessage='100% On-Chain'
                    id='swap.carousel.title.page3'
                  />
                </Title>
                <BodyText weight={500}>
                  <FormattedMessage
                    defaultMessage='All Swap trades are confirmed and settled directly on-chain.'
                    id='swap.carousel.desc.page3'
                  />
                </BodyText>
              </Container>
            </CarouselPageWrapper>
          </div>
          <div>
            <CarouselPageWrapper>
              <Container>
                <Title size='24px' weight={600}>
                  <FormattedMessage
                    defaultMessage='You Control Your Key'
                    id='swap.carousel.title.page4'
                  />
                </Title>
                <BodyText weight={500}>
                  <FormattedMessage
                    defaultMessage='With Swap your crypto is safe, secure, and your keys are always intact.'
                    id='swap.carousel.desc.page4'
                  />
                </BodyText>
              </Container>
            </CarouselPageWrapper>
          </div>
          <div>
            <CarouselPageWrapper>
              <Container>
                <Title size='24px' weight={600}>
                  <FormattedMessage
                    defaultMessage='Manage Risk Better'
                    id='swap.carousel.title.page5'
                  />
                </Title>
                <BodyText weight={500}>
                  <FormattedMessage
                    defaultMessage='Introducing Digital Dollars (USD PAX) to de-risk your crypto investment or lock-in gains.'
                    id='swap.carousel.desc.page5-2'
                  />
                </BodyText>
              </Container>
            </CarouselPageWrapper>
          </div>
        </Slider>
        <PlacementDiv />
        <CarouselImage1 name='swap-carousel1' currentSlide={currentSlide} />
        <CarouselImage2 name='swap-carousel2' currentSlide={currentSlide} />
        <CarouselImage3 name='swap-carousel3' currentSlide={currentSlide} />
        <CarouselImage4 name='swap-carousel4' currentSlide={currentSlide} />
        <CarouselImage5 name='swap-carousel5' currentSlide={currentSlide} />
        <ButtonContainer>
          <StatusBar step={step} />
        </ButtonContainer>
      </CarouselWrapper>
    </Wrapper>
  )
}

const mapStateToProps = state => getData(state)

export default connect(mapStateToProps)(GetStarted)
