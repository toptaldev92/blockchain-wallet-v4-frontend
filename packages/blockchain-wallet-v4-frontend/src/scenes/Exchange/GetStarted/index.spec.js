import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import GetStarted from './index'
import React from 'react'
import toJson from 'enzyme-to-json'

const mockStore = configureMockStore()
const store = mockStore({})

jest.mock('blockchain-info-components', () => ({
  Image: 'image',
  Text: 'text'
}))

jest.mock('./StatusBar', () => 'StatusBar')

describe('StatusBar', () => {
  it('renders correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <GetStarted />
      </Provider>
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
