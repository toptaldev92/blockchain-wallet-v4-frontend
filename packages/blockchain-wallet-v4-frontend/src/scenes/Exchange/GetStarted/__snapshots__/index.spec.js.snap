// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar renders correctly 1`] = `
<GetStarted__Wrapper>
  <GetStarted__Container>
    <GetStarted__Column>
      <GetStarted__Row
        marginBottom="36px"
      >
        <text
          color="brand-primary"
          size="30px"
          weight={600}
        >
          <FormattedMessage
            defaultMessage="A better way to trade crypto"
            id="scenes.exchange.getstarted.title"
            values={Object {}}
          />
        </text>
      </GetStarted__Row>
      <GetStarted__Row
        marginBottom="24px"
      >
        <GetStarted__DarkText>
          <FormattedMessage
            defaultMessage="The faster, smarter way to trade your crypto."
            id="scenes.exchange.getstarted.introducing"
            values={Object {}}
          />
        </GetStarted__DarkText>
      </GetStarted__Row>
      <GetStarted__Row
        marginBottom="8px"
      >
        <GetStarted__DarkText>
          <FormattedMessage
            defaultMessage="Upgrade now to enjoy the following benefits:"
            id="scenes.exchange.getstarted.upgrade"
            values={Object {}}
          />
        </GetStarted__DarkText>
      </GetStarted__Row>
      <GetStarted__Row
        marginBottom="8px"
      >
        <GetStarted__DarkText>
          <GetStarted__PrimaryText>
            <FormattedMessage
              defaultMessage="Lower cost"
              id="scenes.exchange.getstarted.lowercost"
              values={Object {}}
            />
          </GetStarted__PrimaryText>
          <GetStarted__DarkText
            hideOnMobile={true}
          >
             - 
          </GetStarted__DarkText>
          <FormattedMessage
            defaultMessage="Super competitive crypto exchange prices"
            id="scenes.exchange.getstarted.lowercost_description"
            values={Object {}}
          />
        </GetStarted__DarkText>
      </GetStarted__Row>
      <GetStarted__Row
        marginBottom="8px"
      >
        <GetStarted__DarkText>
          <GetStarted__PrimaryText>
            <FormattedMessage
              defaultMessage="Live rates"
              id="scenes.exchange.getstarted.liverates"
              values={Object {}}
            />
          </GetStarted__PrimaryText>
          <GetStarted__DarkText
            hideOnMobile={true}
          >
             - 
          </GetStarted__DarkText>
          <FormattedMessage
            defaultMessage="You always get the most up to date price"
            id="scenes.exchange.getstarted.liverates_Description"
            values={Object {}}
          />
        </GetStarted__DarkText>
      </GetStarted__Row>
      <GetStarted__Row
        marginBottom="32px"
      >
        <GetStarted__DarkText>
          <GetStarted__PrimaryText>
            <FormattedMessage
              defaultMessage="Higher limits"
              id="scenes.exchange.getstarted.higherlimits"
              values={Object {}}
            />
          </GetStarted__PrimaryText>
          <GetStarted__DarkText
            hideOnMobile={true}
          >
             - 
          </GetStarted__DarkText>
          <FormattedMessage
            defaultMessage="Limits from $1,000-$25,000"
            id="scenes.exchange.getstarted.higherlimits_description"
            values={Object {}}
          />
        </GetStarted__DarkText>
      </GetStarted__Row>
      <GetStarted__Row>
        <StatusBar />
      </GetStarted__Row>
    </GetStarted__Column>
    <GetStarted__Column>
      <GetStarted__PreviewImage />
    </GetStarted__Column>
  </GetStarted__Container>
</GetStarted__Wrapper>
`;
